app/src/main/java/com/example/fittracker/mvp/view/MainScreenView.java

```diff
@@ -138,12 +135,15 @@ public void contractWeatherCard() {
         humidityView.setVisibility(View.GONE);
         pressureTag.setVisibility(View.GONE);
         pressureView.setVisibility(View.GONE);
-        isWeatherCardExpanded = !isWeatherCardExpanded;
     }
 
     @Override
     public boolean isWeatherCardExpanded() {
-        return isWeatherCardExpanded;
+        if (humidityView.getVisibility() == View.GONE) {
+            return false;
+        } else {
+            return true;
+        }
```

Comment:
replace this by

return humidityView.getVisibility() != View.GONE
or
return humidityView.getVisibility() == View.VISIBLE

Reply:
Done